### PR TITLE
Implement methods for NFT module

### LINKED ISSUE #8543: Implement method transfer for NFT module
### Description

Implement method `transfer` as described in the [LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md#transfer-2)

### Acceptance Criteria

  - Add unit tests covering all the path.
  - Stub should not be used for stores

### Additional Information
@mjerkov should be added as a reviewer for this issue

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #8543 #8544 #8585 #8586 

### How was it solved?

- Implement methods `transfer` and `transferCrossChain` as specified
- Resolve circular dependency by adding interface
- Register & init commands in module

### How was it tested?

Added unit tests

### CODE REVIEW

File: framework/src/modules/nft/method.ts [hunk 1]
```diff
@@ -420,4 +433,196 @@ export class NFTMethod extends BaseMethod {
...
+					recipientAddress,
+					receivingChainID,
+					nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+			throw new Error('NFT is escrowed to another chain');
+		}
+
```
Comment (mjerkov) [inline framework/src/modules/nft/method.ts:560]: It seems to me that check if `chainID` is equal to either the `OWN_CHAIN_ID` or `receivingChainID` is missing, see [this part](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md?plain=1#L1851-L1853) of the `transferCrossChain` verification in the LIP.

File: framework/test/unit/modules/nft/method.spec.ts [hunk 2]
```diff
@@ -760,4 +741,287 @@ describe('NFTMethod', () => {
...
+					senderAddress,
+					recipientAddress,
+					receivingChainID,
+					nftID: escrowedNFT.nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+		});
+
```
Comment (mjerkov) [inline framework/test/unit/modules/nft/method.spec.ts:905]: Similarly as in comment for `method.ts`, it seems to me that the unit test for the case when `chainID` is equal to either the `OWN_CHAIN_ID` or `receivingChainID` is missing, see [this part](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md?plain=1#L1851-L1853) of the `transferCrossChain` verification in the LIP.

**Review by mjerkov — CHANGES_REQUESTED** [review]
Nice, I left just two comments about the missing case in `transferCrossChain` verification and related unit test.

File: framework/src/modules/nft/method.ts (hunk 1 above)
Comment (Incede) [reply framework/src/modules/nft/method.ts:560]: indeed. Thanks for noticing this.

**Review by mjerkov — APPROVED** [review]

**Review by has5aan — APPROVED** [review]

**Pull request merged**